### PR TITLE
Define Button's value also as a ReactNode

### PR DESCRIPTION
**Pull request by xMort**
JIRA: TNT-641

---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

File: libs/sdk-ui-filters/src/DateFilter/index.ts [hunk 1]
```diff
@@ -74,3 +74,17 @@ export {
 
 export { defaultDateFilterOptions } from "./constants/config";
 export { GranularityIntlKey } from "./constants/i18n";
+
+// internal parts of DateFilter reused in AD DateFilter component
```
Comment (kandl) [inline libs/sdk-ui-filters/src/DateFilter/index.ts:78]: I'm not sure if we should expose it, if it's marked as internal. I have some bad experience with this, that customers are then importing this internal stuff. IMO full imports in AD are a better option.

Comment (xMort) [reply libs/sdk-ui-filters/src/DateFilter/index.ts]: Fixed. I removed the reexports and left just @internal tag on the components that will be reused by AD.

File: libs/sdk-ui-kit/src/Button/typings.ts [hunk 2]
```diff
@@ -10,7 +12,7 @@ export interface IButtonProps {
     tagName?: string;
     title?: string;
     type?: string;
-    value?: string;
+    value?: string | ReactNode;
```
[the remark this answers is not in the document]
Comment (xMort) [reply libs/sdk-ui-kit/src/Button/typings.ts]: Good point. I checked the React typings and changed it to `ReactNode`.

**Review by kandl — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2447

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2201/ : SUCCESS in 1m 15s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2489/ : SUCCESS in 21s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1313/ : SUCCESS in 17m 32s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9464/ : SUCCESS in 14m 35s

**Pull request merged**